File: benchmark/nodejs/bsearch.js
const start = Date.now();
const arr = [];
for (let i = 0; i < 100000; i++) {
    arr.push(i * 2);
}
let found = 0;
for (let j = 0; j < 10000; j++) {
    const target = j * 20;
    let left = 0;
    let right = 99999;
    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        if (arr[mid] === target) {
            found++;
            break;
        }
        if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
}
const end = Date.now();
console.log(`Found: ${found}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/bsearch.rb
start = Time.now.to_f * 1000
arr = (0...100000).map { |i| i * 2 }
found = 0
(0...10000).each do |j|
  target = j * 20
  left = 0
  right = 99999
  while left <= right
    mid = (left + right) / 2
    if arr[mid] == target
      found += 1
      break
    end
    if arr[mid] < target
      left = mid + 1
    else
      right = mid - 1
    end
  end
end
finish = Time.now.to_f * 1000
puts "Found: #{found}"
puts "Time: #{(finish - start).round}ms"

